### PR TITLE
Changes to behaviour of the Covid class

### PR DESCRIPTION
**Pull request by paulbessell**
Some pre-amble to being able to change the disease pathway structure. Change the distrubution used for the disease periods and changing from days to hours in deteriming the period durations

### CODE REVIEW

**Comment by BlairArchibald** [comment]
Looks good - much clearer having the parameters in hours!

**Comment by paulbessell** [comment]
So this is the final version of changes to this for now. In the new format we might need to go back and look again at the mortality rates, they are a little low, but that is for me to do later

File: src/main/java/uk/co/ramp/covid/simulation/Covid.java [hunk 1]
```diff
@@ -67,61 +82,99 @@ public boolean isRecovered() {
...
+       
         }
     }
 
-    // Cycle through the infection for that timestep
     public CStatus stepInfection() {
-        this.infCounter++;
+    	CStatus status = null;
+    	if(symptomaticCase) status = this.stepInfectionSymptomatic();
+    	if(!symptomaticCase) status = this.stepInfectionAsymptomatic();
```
Comment (bobtaylor12) [inline src/main/java/uk/co/ramp/covid/simulation/Covid.java:111]: Use an if...else clause here

File: src/main/java/uk/co/ramp/covid/simulation/Covid.java [hunk 2]
```diff
@@ -67,61 +82,99 @@ public boolean isRecovered() {
...
-
-        p1 = rng.nextPoisson(meanP1);
-
-        if (ccase.avoidsPhase2(rng.nextUniform(0, 1))) {
-            p2 = 0;
-        } else {
-            p2 = rng.nextPoisson(meanP2);
+        latentPeriod = (double) Math.exp(rng.nextGaussian(Math.log(meanLatentPeriod), 1.0));
+        if(!symptomaticCase) asymptomaticPeriod = Math.exp(rng.nextGaussian(Math.log(meanAsymptomaticPeriod), 1.0));
+        if(symptomaticCase) {
```
Comment (bobtaylor12) [inline src/main/java/uk/co/ramp/covid/simulation/Covid.java:94]: Use if...else clause here

File: src/main/java/uk/co/ramp/covid/simulation/Covid.java (hunk 1 above)
Comment (paulbessell) [reply src/main/java/uk/co/ramp/covid/simulation/Covid.java]: Done

File: src/main/java/uk/co/ramp/covid/simulation/Covid.java (hunk 2 above)
Comment (paulbessell) [reply src/main/java/uk/co/ramp/covid/simulation/Covid.java]: Done

File: src/test/java/uk/co/ramp/covid/simulation/CovidTest.java [hunk 3]
```diff
@@ -14,37 +14,37 @@
...
         //Test that the person becomes phase1 after the asymptomatic period
-        double asymptomaticPeriod = virus.getAsymptomaticPeriod() * 24;
-        for (int i = 0; i < asymptomaticPeriod; i++) {
+        double p1Period = virus.getP1();
+
+        for (int i = 0; i < p1Period; i++) {
             cStatus = virus.stepInfection();
         }
         assertEquals(CStatus.PHASE1, cStatus);
+        assertEquals(true, virus.getIsSymptomatic());
```
Comment (BlairArchibald) [inline src/test/java/uk/co/ramp/covid/simulation/CovidTest.java:40]: There's an "assertTrue" you can use for this

File: src/test/java/uk/co/ramp/covid/simulation/CovidTest.java [hunk 4]
```diff
@@ -56,18 +56,52 @@ public void testStepInfectionRecover() throws IOException {
...
+            assertEquals(false, virus.getIsSymptomatic());
+        	cStatus = virus.stepInfection();
+        }
+        
+        for (int i = 0; i < asymptomaticPeriod; i++) {
+            assertEquals(CStatus.ASYMPTOMATIC, cStatus);
+            assertEquals(false, virus.getIsSymptomatic());
+        	cStatus = virus.stepInfection();
+        }
+//        cStatus = virus.stepInfection();
```
Comment (BlairArchibald) [inline src/test/java/uk/co/ramp/covid/simulation/CovidTest.java:100]: Probably best to remove this so it doesn't confuse people whether it should be there or not!

File: src/test/java/uk/co/ramp/covid/simulation/CovidTest.java (hunk 3 above)
Comment (paulbessell) [reply src/test/java/uk/co/ramp/covid/simulation/CovidTest.java]: Done

File: src/test/java/uk/co/ramp/covid/simulation/CovidTest.java (hunk 4 above)
Comment (paulbessell) [reply src/test/java/uk/co/ramp/covid/simulation/CovidTest.java]: Done

**Pull request merged**